src/import/ImportRaw.cpp

```diff
@@ -496,6 +492,52 @@ void ImportRawDialog::OnPlay(wxCommandEvent & WXUNUSED(event))
 {
 }
 
+void ImportRawDialog::OnDetect(wxCommandEvent & event)
+{
+   try {
+      // Yes, FormatClassifier currently handles filenames in UTF8 format only, that's a TODO ...
+      FormatClassifier theClassifier(mFileName.utf8_str());
+      mEncoding = theClassifier.GetResultFormatLibSndfile();
+      mChannels = theClassifier.GetResultChannels();
+   } catch (...) {
+      // Something went wrong in FormatClassifier, abort.
+      return;
+   }
+
+   int selection = 0;
+   for (int i = 0; i < mNumEncodings; i++) {
+      int subtype = mEncodingSubtype[i];
+      if ((mEncoding & SF_FORMAT_SUBMASK) == subtype) {
+         selection = i;
+         break;
+      }
+   }
+
+   int endian = 0;
+   switch (mEncoding & SF_FORMAT_ENDMASK)
```

Comment:
Make a function of this switch and eliminate the duplication with the constructor of the dialog.

Reply:
Done